flocker/control/_persistence.py

```diff
@@ -0,0 +1,92 @@
+# Copyright Hybrid Logic Ltd.  See LICENSE file for details.
+
+"""
+Persistence of cluster configuration.
+"""
+
+from pickle import dumps, loads
+
+from twisted.application.service import Service
+from twisted.internet.defer import succeed
+
+from ._model import Deployment
+
+
+# These should not use Pickle!@!
+# https://clusterhq.atlassian.net/browse/FLOC-1241x
+def serialize_deployment(deployment):
+    """
+    Convert a ``Deployment`` object to ``bytes``.
+
+    :param Deployment deployment: Object to serialize.
+
+    :return bytes: Serialized object.
+    """
+    return dumps(deployment)
+
+
+def deserialize_deployment(data):
+    """
+    Create a ``Deployment`` object that was previously serialized to given
+    ``bytes``.
+
+    :param bytes data: Output of ``serialize_deployment``.
+
+    :return Deployment: Deserialized object.
+    """
+    return loads(data)
+
+
+class ConfigurationPersistenceService(Service):
+    """
+    Persist configuration to disk, and load it back.
+
+    :ivar Deployment _deployment: The current desired deployment configuration.
+    """
+    def __init__(self, reactor, path):
+        """
+        :param reactor: Reactor to use for thread pool.
+        :param FilePath path: Directory where desired deployment will be
+            persisted.
+        """
+        self._path = path
+
+    def startService(self):
+        if not self._path.exists():
+            self._path.makedirs()
+        self._config_path = self._path.child(b"current_configuration.json")
+        if self._config_path.exists():
+            self._deployment = deserialize_deployment(
+                self._config_path.getContent())
+        else:
+            self._deployment = Deployment(nodes=frozenset())
+            self._sync_save(self._deployment)
```

Comment:
Worry about it when we get there. Maybe I'll finally have a reason to finish that dependency injection framework.